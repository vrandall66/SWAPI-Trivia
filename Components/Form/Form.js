import React from "react";

// render() {
// return (
//   <form>
//     <input type="text" name="name" placeholder="name" />
//     <input type="text" name="quote" placeholder="Favorite quote from the films" />
//     <select name="ranking">
//      <option value="novice">Novice</option>
//      <option value="intermediate">Intermediate</option>
//      <option value="expert"><Expert</option>
//     </select>
//     <button type="button">Submit</button>
//    </form>
