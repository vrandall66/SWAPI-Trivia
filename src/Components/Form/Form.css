form {
  align-items: center;
  border-radius: 180%;
  display: flex;
  flex-direction: column;
  font-family: 'Roboto Condensed', sans-serif;
  height: 75vh;
  justify-content: center;
  margin: 10px;
  padding: 10px;
  text-transform: uppercase;
  width: 75vw;
}

input,
select {
  margin-bottom: 1%;
}