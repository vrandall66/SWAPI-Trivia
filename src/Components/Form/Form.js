import React from "react";
import { Link, withRouter } from "react-router-dom";
import "./Form.css";

export class Form extends React.Component {
  constructor() {
    super();
    this.state = {
      name: "",
      favQuote: "",
      ranking: "Novice",
      error: false
    };
  }

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  handleSubmit = () => {
    if (this.state.name.length && this.state.favQuote.length) {
      this.props.handleFormSubmit(this.state);
      this.props.history.push("/movies");
    } else {
      this.setState({ error: true });
    }
  };

  render() {
    return (
      <form className="Form">
        <label htmlFor="name" className="UserModalFormLabel">
          Name
        </label>
        <input
          className="UserModalFormInput"
          id="name"
          type="text"
          name="name"
          placeholder="Name"
          onChange={this.handleChange}
          value={this.state.name}
        />
        <label htmlFor="favQuote" className="UserModalFormLabel">
          Favorite Quote
        </label>
        <input
          className="UserModalFormInput"
          id="favQuote"
          type="text"
          name="favQuote"
          placeholder="Favorite quote from the films"
          onChange={this.handleChange}
          value={this.state.favQuote}
        />
        <label htmlFor="ranking" className="UserModalFormLabel">
          Select Fan Ranking
        </label>
        <select
          id="ranking"
          name="ranking"
          className="UserModalFormSelect"
          onChange={this.handleChange}
        >
          <option value="Novice">Novice</option>
          <option value="Intermediate">Intermediate</option>
          <option value="Expert">Expert</option>
        </select>
        <button
          type="button"
          className="UserModalFormBtn"
          onClick={() => this.handleSubmit()}
        >
          Submit
        </button>
        {this.state.error && (
          <h3>
            The Force Is Not With You! <br /> Please Complete The Form
          </h3>
        )}
      </form>
    );
  }
}

export default withRouter(Form);
