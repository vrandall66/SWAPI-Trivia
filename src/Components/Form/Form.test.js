import React from "react";
import { shallow } from "enzyme";
import Form from "./Form";

describe("Form", () => {
  let wrapper;
  const mockSearchFunc = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Form handleFormSubmit={mockSearchFunc} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should update state.name on keydown of name input", () => {
    const mockEvent = {
      target: {
        name: "name",
        value: "Matt"
      }
    };

    const mockEvent2 = {
      target: {
        name: "favQuote",
        value: "I know"
      }
    };

    wrapper.instance().handleChange(mockEvent);
    wrapper.instance().handleChange(mockEvent2);

    expect(wrapper.state()).toEqual({
      name: "Matt",
      favQuote: "I know",
      error: false,
      ranking: "novice"
    });
  });
});
