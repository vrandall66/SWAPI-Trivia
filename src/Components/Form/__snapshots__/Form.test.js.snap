// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<form
  className="Form"
>
  <label
    className="UserModalFormLabel"
    htmlFor="name"
  >
    Name
  </label>
  <input
    className="UserModalFormInput"
    id="name"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <label
    className="UserModalFormLabel"
    htmlFor="favQuote"
  >
    Favorite Quote
  </label>
  <input
    className="UserModalFormInput"
    id="favQuote"
    name="favQuote"
    onChange={[Function]}
    placeholder="Favorite quote from the films"
    type="text"
    value=""
  />
  <label
    className="UserModalFormLabel"
    htmlFor="ranking"
  >
    Select Fan Ranking
  </label>
  <select
    className="UserModalFormSelect"
    id="ranking"
    name="ranking"
    onChange={[Function]}
  >
    <option
      value="novice"
    >
      Novice
    </option>
    <option
      value="intermediate"
    >
      Intermediate
    </option>
    <option
      value="expert"
    >
      Expert
    </option>
  </select>
  <Link
    to="/movies"
  >
    <button
      className="UserModalFormBtn"
      onClick={[Function]}
      type="button"
    >
      Submit
    </button>
  </Link>
</form>
`;
